tracer/src/Datadog.Trace/Processors/Obfuscator.cs

```diff
@@ -0,0 +1,177 @@
+                        if (char.IsDigit(sqlChars[sequenceStart]))
+                        {
+                            sqlChars[sequenceStart] = '?';
+                            modified = true;
+                        }
+                    }
+                    else if (sequenceStart < sequenceEnd)
+                    {
+                        if (IsQuoted(sqlChars, sequenceStart, sequenceEnd)
+                            || IsNumericLiteralPrefix(sqlChars[sequenceStart])
+                            || IsHexLiteralPrefix(sqlChars, sequenceStart, sequenceEnd))
+                        {
+                            var length = sequenceEnd - sequenceStart;
+                            Array.Copy(sqlChars, end, sqlChars, sequenceStart + 1, outputLength - end);
+                            sqlChars[sequenceStart] = '?';
+                            outputLength -= length;
+                            modified = true;
+                        }
+                    }
+
+                    end = start;
+                    start = PreviousSetBit(splitterBytes, start - 1);
+                }
+
+                if (modified)
+                {
+                    var charArray = new char[outputLength];
+                    Array.Copy(sqlChars, charArray, outputLength);
+
+                    return new string(charArray);
+                }
+            }
+            catch (Exception ex)
+            {
+                Log.Debug("Error obfuscating sql {}", sql, ex);
+            }
+
+            return sql;
+        }
+
+        private static BitArray FindSplitterPositions(char[] sqlChars)
+        {
+            var positions = new BitArray(sqlChars.Length);
+
+            var quoted = false;
+            var escaped = false;
+
+            for (var i = 0; i < sqlChars.Length; ++i)
+            {
+                var c = sqlChars[i];
+                if (c == '\'' && !escaped)
+                {
+                    quoted = !quoted;
+                }
+                else
+                {
+                    escaped = (c == '\\') & !escaped;
+                    positions.Set(i, !quoted & IsSplitter(c));
+                }
+            }
+
+            return positions;
+        }
+
+        private static bool IsSplitter(char c)
+        {
+            if (Convert.ToInt16(c) < 256)
+            {
+                return Splitters.Get(c);
+            }
+
+            return false;
+        }
+
+        private static bool IsNumericLiteralPrefix(char c)
+        {
+            return NumericLiteralPrefix.Get(Convert.ToByte(c));
+        }
+
+        private static bool IsQuoted(char[] sqlChars, int start, int end)
+        {
+            return (sqlChars[start] == '\'' && sqlChars[end] == '\'');
+        }
+
+        private static bool IsHexLiteralPrefix(char[] sqlChars, int start, int end)
+        {
+            return (sqlChars[start] | ' ') == 'x' && start + 1 < end && sqlChars[start + 1] == '\'';
```

Comment:
It took me a while to figure out what the `| ' '` part did, so I think it would be better with a comment:

```suggestion
            // | 0x20 converts ASCII characters to lowercase
            return (sqlChars[start] | 0x20) == 'x' && start + 1 < end && sqlChars[start + 1] == '\'';
```